### PR TITLE
Add certificate requirements for OAuth to work

### PR DESCRIPTION
**Pull request by AlekhyaSasi**
Fixes: https://github.com/flyteorg/flyte/issues/2371

### CODE REVIEW

File: rsts/deployment/security/security.rst [hunk 1]
```diff
@@ -43,3 +43,18 @@ context for these.
...
+
+************
+OAuth
+************
+Flytectl needs a CA-certified SSL cert for OAuth to work. If a self-signed certificate is used, the following error is raised.
+
+.. code-block::
+    
+    certificate is not standards compliant.
+
```
Comment (SmritiSatyanV) [inline rsts/deployment/security/security.rst:56]: Flytectl requires CA-certified SSL cert for OAuth to work. Using a self-signed certificate throws the error:

**Review by kumare3 — COMMENTED** [review]
cc @EngHabu (lgtm)

File: rsts/deployment/security/security.rst [hunk 2]
```diff
@@ -4,42 +4,57 @@
 Security Overview
 ###################
 
-Here we cover security aspects of running your flyte deployments. In the current state we will be covering the user
-which is used for running the flyte services and we will go through why we do this and not run them as root user
+Here we cover the security aspects of running your flyte deployments. In the current stat, we will cover the user
```
Comment (samhita-alla) [inline rsts/deployment/security/security.rst:7]:
```suggestion
Here we cover the security aspects of running your flyte deployments. In the current stat,e we will cover the user
```

Comment (EngHabu) [reply rsts/deployment/security/security.rst:7]: +1

**Review by EngHabu — COMMENTED** [review]
Edits

**Review by samhita-alla — APPROVED** [review]

**Pull request merged**